app/models/accounting/quickbooks/transaction_creator.rb

```diff
@@ -0,0 +1,78 @@
+module Accounting
+  module Quickbooks
+    class TransactionCreator
+      attr_reader :qb_connection
+
+      def initialize(qb_connection = Division.root.qb_connection)
+        @qb_connection = qb_connection
+      end
+
+      def add_disbursement(amount:, loan_id:, memo:, qb_bank_account_id:, qb_customer_id:)
+        je = ::Quickbooks::Model::JournalEntry.new
+        je.private_note = memo
+
+        je.line_items << create_line_item(
+          amount: amount,
+          loan_id: loan_id,
+          posting_type: 'Debit',
+          qb_account_id: Division.root.interest_receivable_account.qb_id,
+          qb_customer_id: qb_customer_id
+        )
+
+        je.line_items << create_line_item(
+          amount: amount,
+          loan_id: loan_id,
+          posting_type: 'Credit',
+          qb_account_id: qb_bank_account_id,
+          qb_customer_id: qb_customer_id
+        )
+
+        service = ::Quickbooks::Service::JournalEntry.new(qb_connection.auth_details)
+
+        service.create(je)
+      end
+
+      private
+
+      def create_line_item(amount:, loan_id:, posting_type:, qb_account_id:, qb_customer_id:)
+        line_item = ::Quickbooks::Model::Line.new
+        line_item.detail_type = 'JournalEntryLineDetail'
+        jel = ::Quickbooks::Model::JournalEntryLineDetail.new
+        line_item.journal_entry_line_detail = jel
+
+        jel.entity = create_customer(qb_customer_id: qb_customer_id)
+
+        # The QBO api needs a fully persisted class before we can associate it.
+        # We need to either find or create the class, and use the returned Id.
+        jel.class_id = find_or_create_qb_class(loan_id: loan_id).id
+
+        line_item.amount = amount
+        jel.posting_type = posting_type
+        jel.account_id = qb_account_id
+
+        line_item
+      end
+
+      def create_customer(qb_customer_id:)
```

Comment:
The name here seems a bit misleading. You're not actually creating a new customer in QB for every line item are you? Just kind of creating an entity object for them? Perhaps `get_customer` would be better? Also couldn't this happen just once in `add_disbursement` and then you just pass the entity to create line item instead of the customer_id?